cli/src/config/templates.toml

```diff
@@ -16,10 +16,10 @@ commit_summary = 'format_commit_summary_with_refs(self, bookmarks)'
 
 annotate_commit_summary = '''
 separate(" ",
-  format_short_id(change_id),
-  format_short_id(commit_id),
-  format_short_signature(author),
-  format_timestamp(committer.timestamp()),
+  change_id.shortest(8),
+  commit_id.shortest(8),
+  pad_end(8, truncate_end(8, author.username())),
```

Comment:
Perhaps `author.email()` would be better (even though I had just the username in my screenshot), in case we have somebody with `me@one-cool-username.com` and another person with `me@another-cool-username`.

If you disagree, I'd use `pad(truncate(author.username() ++ '@'))`, to make it clearer what this field is.

More questionably, after looking at the tests, I'm not sure whether 8 characters is enough, I think we could try 10. We can also keep it as is and change it later, if needed.

Reply:
This is not solely related to this PR, but I dislike how we have an entire field in commits that represents an appropriate human‐readable identifier to use for their authors and committers but ignore it to just show email addresses instead, and then run into situations like this where we’re hoping that we can use the local‐part as a sensible global identifier. It makes sense in a corporate context where everyone is `‹username›@google.com`, but not for collaborative projects in general.

Of course I’m biased since the local‐parts of my email addresses don’t identify me at all.

Reply:
`username()` (which is sometimes shorter than 8 chars) looked nicer than full 8 or 10 `email()` characters truncated at arbitrary place. I don't have a strong opinion, though.

I wouldn't add `@` suffix unless we do the same for `builtin_log_onenline`.

Reply:
For reference, here is the log with emails trimmed to 10 chars. I also make the `@` white somewhat indulgently (it'd be easier if we supported `signature.domain()`).

![image](https://github.com/user-attachments/assets/921a4ec4-90fb-4a7c-ad16-eb905e961230)

Reply:
I'm not sure what people other than me prefer. Is a trimmed email a worthwhile compromise between a trimmed username and a full name trimmed to, say, 20 characters?

> I wouldn't add @ suffix unless we do the same for builtin_log_onenline.

I wouldn't mind adding `@` to `builtin_log_onenline` as well, especially if we don't go with the full email here and adding it to the username was conditional on adding it to that template. But again, this might be my personal aesthetics.

Reply:
I also don't feel very strongly about this, by the way. I can always configure it and, more importantly, we can try something and see how/whether people react.

Reply:
Tig supports an abbreviated author name that seems to split the name by spaces, use initials, and then the full last name. Examples from `tig blame Cargo.toml`:

- ASeipp
- MvZweigbergk
- EMesterhazy
- IGrigoriev
- db   (from `dependabot[bot]`)
- SJennings
- YNishihara
- BTan
- JOkoński 
- STardieu

I'm not sure we're lacking in options, but I thought this is worth mentioning.

Reply:
Discussion continues to https://github.com/martinvonz/jj/pull/4653#discussion_r1816545416, closing as resolved.